enhancements/console/dynamic-plugins.md

```diff
@@ -0,0 +1,389 @@
+show a checkbox to enable plugins during operator install. The value is a
+serialized JSON array of strings, where each item is the name of a
+`ConsolePlugin` resource the operator will create.
+
+```yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    console.openshift.io/plugins: '["my-plugin"]'
+```
+
+For operators from the `redhat-operators` catalog source, the install operator
+page will check the plugin checkbox by default. For other operators, the
+checkbox will not be checked by default. Any plugin can be enabled or disabled
+later through the console cluster settings page.
+
+### Loading Plugins
+
+The console backend will proxy requests to the plugin `Service`. This way the
+`Service` itself does not need to be exposed outside the cluster, and avoids
+any same-origin or certificate errors in the browser loading JavaScript from a
+separate domain.
+
+When loaded in the browser, console will read the manifest for each plugin. The
+manifest is a JSON document that contains metadata about the plugin and the
+extensions. `extensions` mirrors the API used for static plugins, and the same
+extension points will be available.
+
+```json
+{
+  "$schema": "/path/to/schema/manifest.json",
+  "id": "my-plugin",
+  "version": "1.2.3",
+  "pluginAPI": "4.7.x",
+  "displayName": "My Plugin",
+  "extensions": [
+    {
+      "type": "console.flag",
+      "properties": {
+        "handler": {
+          "$codeRef": "example.testHandler"
+        }
+      }
+    },
+    {
+      "type": "console.flag/model",
+      "properties": {
+        "flag": "EXAMPLE",
+        "model": {
+          "group": "example.com",
+          "version": "v1",
+          "kind": "ExampleModel"
+        }
+      }
+    }
+  ]
+}
+```
+
+An extension may contain code references, encoded as object literal
+`{ $codeRef: string }`. The value of `$codeRef` is `moduleName.exportName` or
+`moduleName` (equivalent to `moduleName.default`). Webpack itself will resolve
+the remote module when console calls `container.get` on the
+[remote container](https://webpack.js.org/concepts/module-federation/#dynamic-remote-containers).
+
+#### Plugin Manifest Properties
+
+| Property | Description |
+| ----- | -------- |
+| `name: string` | Used as a unique identifier. Each plugin must have a unique name. |
+| `version: string` | The version of the plugin. Version must be parsable by node-semver, eg. “1.2.3” |
+| `pluginAPI: string` | Semver range of compatible OpenShift console versions. |
+| `displayName?: string` | User friendly display name. |
+| `description?: string` | The description of the plugin. |
+| `extensions: Extension[]` | List of extensions contributed by the plugin. |
+| `dependencies?: { [pluginName: string]: string }` | A plugin which depends on other plugins must explicitly define their dependencies. The key of the map is the plugin name and the value is a semver version range. |
```

Comment:
what happens if a dependency is not met?

Reply:
The plugin isn't loaded.

Reply:
is that stated somewhere in here?

Reply:
Yeah, I pushed an update this morning to call that out.